File: backend/app/schemas/resource.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import ResourceCategory, ResourceStage


class ResourceResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    title: str
    summary: str
    category: ResourceCategory
    stage: ResourceStage
    source_name: str
    source_url: str
    created_at: datetime
    updated_at: datetime


class ResourceDetailResponse(ResourceResponse):
    """The single-resource response, used only by GET /resources/{id}.

    Includes the full article body -- deliberately excluded from
    ResourceResponse/GET /resources so the list endpoint stays lightweight
    as the catalog grows.
    """

    body: str

File: lib/features/information/data/resource.dart
class Resource {
  final String id;
  final String title;
  final String summary;
  final String category;
  final String stage;
  final String sourceName;
  final String sourceUrl;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Resource({
    required this.id,
    required this.title,
    required this.summary,
    required this.category,
    required this.stage,
    required this.sourceName,
    required this.sourceUrl,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Resource.fromJson(Map<String, dynamic> json) {
    return Resource(
      id: json['id'] as String,
      title: json['title'] as String,
      summary: json['summary'] as String,
      category: json['category'] as String,
      stage: json['stage'] as String,
      sourceName: json['source_name'] as String,
      sourceUrl: json['source_url'] as String,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }
}
